### PR TITLE
Restructure public cloud deploy section

### PR DESCRIPTION
**Pull request by jdreinhardt**
Moved all pages related to deploying in a public cloud under a single point in the deploying section. 

Minor tweaks to text to fix links and routing.

### CODE REVIEW

File: docs/02.deploying/06.public-cloud/03.marketplace-faq/03.marketplace-faq.md [hunk 1]
```diff
@@ -260,11 +257,11 @@ NeuVector is updated with major, minor, or patch versions every 6-8 weeks. To up
 
 - For AWS:
 
-  - You can find out more about the NeuVector Prime AWS Marketplace listing in the [NeuVector documentation](../deploying/awsmarketplace).
+  - You can find out more about the NeuVector Prime AWS Marketplace listing in the [NeuVector documentation](../deploying/public-cloud/awsmarketplace).
 
 - For Azure:
 
-  - You can find out more about the NeuVector Prime AWS Marketplace listing in the [NeuVector documentation](../deploying/azuremarketplace).
+  - You can find out more about the NeuVector Prime AWS Marketplace listing in the [NeuVector documentation](../deploying/public-cloud/azuremarketplace).
```
Comment (nunix) [inline docs/02.deploying/06.public-cloud/03.marketplace-faq/03.marketplace-faq.md:264]:
```suggestion
  - You can find out more about the NeuVector Prime AWS Marketplace listing in the [NeuVector documentation](azuremarketplace).
```

File: docs/02.deploying/06.public-cloud/03.marketplace-faq/03.marketplace-faq.md [hunk 2]
```diff
@@ -260,11 +257,11 @@ NeuVector is updated with major, minor, or patch versions every 6-8 weeks. To up
 
 - For AWS:
 
-  - You can find out more about the NeuVector Prime AWS Marketplace listing in the [NeuVector documentation](../deploying/awsmarketplace).
+  - You can find out more about the NeuVector Prime AWS Marketplace listing in the [NeuVector documentation](../deploying/public-cloud/awsmarketplace).
```
Comment (nunix) [inline docs/02.deploying/06.public-cloud/03.marketplace-faq/03.marketplace-faq.md:260]:
```suggestion
  - You can find out more about the NeuVector Prime AWS Marketplace listing in the [NeuVector documentation](awsmarketplace).
```

**Review by nunix — COMMENTED** [review]
Adapted the links for the marketplaces.
As it's at the "same level" in terms of browsing, there's no need to go back in hierarchy.

However, as there's the additional level for "public-cloud", the following links need to be adapted too with an additional "../"
Line 186: (../../updating/updating)
Line 196: (../../navigation/multicluster)
Line 205: (../../navigation/multicluster)
Line 218: (../../deploying/production#backups-and-persistent-data)
Line 218: (../../deploying/restore)
Line 252: (../../updating/updating)

**Comment by jdreinhardt** [comment]
Thank you for the note. I have updated the link levels as noted in the latest commit.

**Review by sunilarjun — APPROVED** [review]
Thank you!

**Pull request merged**